tests/polynomial/ops.mlir

```diff
@@ -68,4 +73,14 @@ module {
     %1 = polynomial.constant <1 + x**2> : !polynomial.polynomial<#ring1>
     return
   }
+
+  func.func @test_ntt(%0 : !poly_ty) {
+    %1 = polynomial.ntt %0 : !poly_ty -> tensor<1024xi32, #ring>
+    return
+  }
+
+  func.func @test_intt(%0 : tensor<1024xi32, #ring>) {
+    %1 = polynomial.intt %0 : tensor<1024xi32, #ring> -> !poly_ty
+    return
+  }
```

Comment:
I think if you want to check for errors it is cleaner to split the error-checking into a separate file, because `--verify-diagnostics` is somewhat incompatible with `FileCheck`, or at least it's hard to get it right because `--verify-diagnostics` will produce exit code 1 on failure, which messes with how `lit` continues the pipe.